### PR TITLE
Patch 22394

### PR DESCRIPTION
**Pull request by jasonaden**
A patch of PR #22394

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/router/src/router.ts [hunk 1]
```diff
@@ -272,6 +272,23 @@ export class Router {
...
+   * Defines when the router updates the browser URL. The default behavior is to update after
+   * successful navigation. However, some applications may prefer a mode where the URL gets
+   * updated at the beginning of navigation. The most common use case would be updating the
+   * URL early so if navigation fails, you can show an error message with the URL that failed.
+   * Available options are:
+   *
+   * - `'deferred'`, the default, updates the browser URL after navigation has finished.
+   * - `'eager'`, updates browser URL at the beginning of navigation.
+   */
+  urlUpdateStrategy: 'deferred'|'eager' = 'deferred';
```
Comment (vicb) [inline packages/router/src/router.ts]: This should not be part of the patch
